src/dmd/escape.d

```diff
@@ -309,24 +309,27 @@ bool checkParamArgumentEscape(Scope* sc, FuncDeclaration fdc, Parameter par, Exp
      */
     void unsafeAssign(VarDeclaration v, const char* desc)
     {
-        if (global.params.useDIP1000 == FeatureState.enabled && sc.func.setUnsafe())
+        if (sc.func.setUnsafe())
```

Comment:
Doesn't this (and the change at L1045) change the semantic of programs not using DIP1000 as well ?
`setUnsafe` has side effect. We had a bug related to it: https://github.com/dlang/dmd/pull/12127/files#diff-1a1c16e5ec319550bcd4ecb5d3f4ca6ca7ffad78af91f077603617117bb7d0c7R1037-R1038

Reply:
Ah, I hadn't considered the short-circuiting nature of the check. Thanks!

Reply:
After checking the code, I think this is fine, since it only has a side-effect when the compiler is determining safety anyway.

Reply:
Some projects on buildkite are currently failing because formerly `@safe` functions are now inferred `@system`, probably because of this `setUnsafe()`